src/QueueManager.php

```diff
@@ -237,5 +263,30 @@ public static function push($className, array $data = [], array $options = []):
 
         $client = static::engine($name);
         $client->sendEvent($queue, $message);
+
+        if (!empty($class::$shouldBeUnique)) {
+            $uniqueId = static::getUniqueId($class, $method, $data);
+
+            Cache::write($uniqueId, true, 'Cake/Queue.queueUnique');
```

Comment:
Should we use the atomic write method `add()` instead? While it doesn't do anything special for file storage for memcache/redis it offers better atomicity.

Reply:
Sure, I'll defer to you on that, I wasn't immediately able to find documentation on `add()`.

Reply:
Oh darn, I didn't know we were missing docs for that method. I'll get that fixed.